### PR TITLE
Cheap bugfixes

### LINKED ISSUE #22343: Make everyone traitors admin button is broken
#### Description of issue
The one in the secrets panel, does fuck all

### LINKED ISSUE #22557: infinite metal rods
(WEB REPORT BY: repertor REMOTE: 206.221.180.138:7777)
# Revision
6478a3d0a48ba64b48ab766d8d11996b1a4ced0b

# Description
creating a full stack of 60 rods makes an empty stack that can be used forever
# Steps to Reproduce
make full stack of rods and make a space bridge
# What you Expected
one stack of finite rods
# What Actually Happened
one stack finite one stack infinite

### LINKED ISSUE #22623: AIs nullspace the cell in the telepad control console when they eject it
#### Description of issue

click on telesci computer
click on eject cell
cell is now in nullspace

probably tries to put it in your hand or something, stupid

#### Difference between expected and actual behavior
be AI
go to telepad control console
click eject
cell pops out of computer
vs
be AI
go to telepad control console
click eject
cell disappears for all eternity

#### Steps to reproduce
be AI
go to telepad control console
click eject

#### Server revision
d93d06b0497bc2f47d4fa4534c4f9c4e69a8e0df

### LINKED ISSUE #22626: Canoot look at role panel of mobs who have been deleted
(WEB REPORT BY: shiftyrail REMOTE: 206.221.180.138:7777)
# Revision
d93d06b0497bc2f47d4fa4534c4f9c4e69a8e0df

# Description

The runtime is : 

```
[16:36:22] Runtime in mind.dm, line 149: Cannot read null.client

client
proc name: role panel (/datum/mind/proc/role_panel)
usr: Arek Tjallson (shiftyrail) (/mob/dead/observer)
usr.loc: The floor (305, 247, 1) (/turf/simulated/floor)
src: Arek Tjallson (/datum/mind)
call stack:
Arek Tjallson (/datum/mind): role panel()
/datum/admins (/datum/admins): Show Role Panel(Arek Tjallson (/mob/dead/observer))
/datum/admins (/datum/admins): Topic("src=\[0x2113a87c];traitor=\[0x...", /list (/list))
Shifty_rail (/client): Topic("src=\[0x2113a87c];traitor=\[0x...", /list (/list), /datum/admins (/datum/admins))
Shifty_rail (/client): Topic("src=\[0x2113a87c];traitor=\[0x...", /list (/list), /datdum/admins (/datum/admins))
```

### LINKED ISSUE #22633: simple mob speech broke
#### Description of issue

they dont speak anymore, just whitespace
tested and it occurs with zombies and other ones that dont make sounds, cats and dogs make their sounds and mice still speak like mice

probably #22351 [bugfix]

### PR DESCRIPTION
**Pull request by ShiftyRail**
[bugfix]
One per commit
Closes #22633 
Closes #22623 
Closes #22557 
Closes #22343 
Closes #22626 

:cl:
- bugfix: Simple mobs will now emote (and speak) properly.
- bugfix: AIs will no longer nullspace the telepad cell when they try to eject it.
- bugfix: You can no longer make infinite stacks of rods or floor tiles.

### CODE REVIEW

**Comment by SGuivenne** [comment]
But muh unatomicness!

**Comment by SonixApache** [comment]
🆑  ?

**Comment by ShiftyRail** [comment]
All of those are minor bugs, is a :cl: needed ?

**Comment by SonixApache** [comment]
¯\_(ツ)_/¯
If someone noticed the bug before they might see that it's fixed in the 🆑 
I mean you might as well, right?

**Comment by Kurfursten** [comment]
1-3 need 🆑 
Admin features do not need CL

use bugfix:

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -164,6 +164,9 @@
 /obj/item/stack/proc/use(var/amount)
 	ASSERT(isnum(src.amount))
 
+	if (src.amount <= 0)
+		qdel(src) // We don't have anything left
```
Comment (PaulRitter) [inline code/game/objects/items/stacks/stack.dm:168]: return after this? idk about qdel

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -320,6 +323,8 @@
 				return S
 	var/obj/item/stack/S = new_stack_type
 	for(var/i = 0 to round(add_amount/initial(S.max_amount)))
+		if (!add_amount)
```
Comment (PaulRitter) [inline code/game/objects/items/stacks/stack.dm]: why not `add_amount <= 0` incase `add_amount < 0`

**Comment by Goldsteel** [comment]
Oversight about the mob speech. Saves me fixing it. I didn't realise picking from an empty list would throw up a runtime.

File: code/modules/telesci/telesci_computer.dm [hunk 3]
```diff
@@ -393,11 +393,14 @@ var/list/telesci_warnings = list(
 
 	if(href_list["eject_cell"])
 		if(cell)
-			usr.put_in_hands(cell)
+			if (usr.put_in_hands(cell))
+				usr.visible_message("<span class='notice'>[usr] removes the cell from \the [src].</span>", "<span class='notice'>You remove the cell from \the [src].</span>")
+			else 
+				visible_message("<span class='notice'>\The [src] beeps as its cell is removed.</span>")
+				cell.forceMove(get_turf(src))
 			cell.add_fingerprint(usr)
```
Comment (PaulRitter) [inline code/modules/telesci/telesci_computer.dm:401]: you would still add a fingerprint for usr's that don't have hands. move this into `if (usr.put_in_hands(cell))`

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: code/modules/telesci/telesci_computer.dm (hunk 3 above)
Comment (ShiftyRail) [reply code/modules/telesci/telesci_computer.dm:401]: Fingerprints for things that don't have hands are stocked in "hidden_fingerprints" and are useful for admin investigation.
Add_fingerprint knows to add it to the correct list of fingerprints

Comment (PaulRitter) [reply code/modules/telesci/telesci_computer.dm:401]: ah i see

**Comment by SonixApache** [comment]
Simple mobs, btw. Holomen are completely unable to speak until this is merged. 🙂

**Pull request merged**